websites/permissions.py

```diff
@@ -0,0 +1,165 @@
+""" permissions for websites """
+from django.contrib.auth.models import Group, Permission
+from django.db import transaction
+from guardian.shortcuts import assign_perm
+from rest_framework.generics import get_object_or_404
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+from websites import constants
+from websites.models import Website
+
+
+def assign_object_permissions(group_or_user, website, perms):
+    """
+    Assign appropriate permissions to a website admin group
+
+    Args:
+        group_or_user (Group or User): the group/user to add permissions to
+        website (Website): The website that the permissions are for
+        perms (list of str): The permissions to be assigned
+    """
+    for perm in perms:
+        try:
+            assign_perm(perm, group_or_user, website)
+        except Permission.DoesNotExist as err:
+            raise Permission.DoesNotExist(f"Permission '{perm}' not found") from err
+
+
+@transaction.atomic
+def create_website_groups(website):
+    """
+    Create groups and assign permissions for a website
+
+    Args:
+        website (Website): The website that the permissions are for
+    """
+    admin_group, _ = Group.objects.get_or_create(
+        name=f"{constants.ADMIN_GROUP}{website.uuid.hex}"
+    )
+    editor_group, _ = Group.objects.get_or_create(
+        name=f"{constants.EDITOR_GROUP}{website.uuid.hex}"
+    )
+    assign_object_permissions(admin_group, website, constants.PERMISSIONS_ADMIN)
+    assign_object_permissions(editor_group, website, constants.PERMISSIONS_EDITOR)
```

Comment:
It seems like this method should be doing what the signal does: check if the website has an owner and add the permissions for that user if there is one. If we run that backpopulate management command, I think we would want to ensure that any existing websites with an owner have the correct permissions assigned for that owner.